packages/hint-manifest-icons/src/hint.ts

```diff
@@ -302,6 +337,21 @@ export default class ManifestIconHint implements IHint {
             }
         };
 
+        const invalidManifest = ({ errors, resource }: ManifestInvalidSchema) => {
+            for (const error of errors) {
+                const isIconError = error.dataPath.match(iconsErrorRegex);
+
+                if (isIconError) {
+                    // Is error the right severity?
+                    context.report(resource, getMessage('iconPurposeInvalid', context.language, isIconError[1]), {
```

Comment:
The [spec says the following](https://w3c.github.io/manifest/#purpose-member):

> If an icon contains multiple purposes, it could be used for any of those purposes. If none of the stated purposes are recognized, the icon is totally ignored. For example, if an icon has purpose "monochrome fizzbuzz", then it could be used as a monochrome icon, as "monochrome" is a valid purpose. However, if an icon just has the purpose "fizzbuzz", then it will be ignored.

That said, I think that we should have an Error if none of the purposes are recognized (the icon is not going to work so that's probably an error) and a warning otherwise.
We could maybe lower the severity one level for each but probably at that moment we will have to look at other icons and what they do, etc. which might complicate things a lot for not a lot of value.

@Malvoz what do you think?

Reply:
If having an invalid value "is not a problem" we should modify the schema to support an string, and do all the checks in the hint.

Reply:
That's what the spec says so...

Reply:
Ok, I will:

1. Remove the regex from the schema.
2. Check if any value is invalid.
3. In case any value is invalid, report a warning.

> We could maybe lower the severity one level for each but probably at that moment we will have to look at other icons and what they do, etc. which might complicate things a lot for not a lot of value.

If you mean the other icons in the list of icons, I think we should report what we found in each icon independently.